src/Async/EventStoreConnection.php

```diff
@@ -184,74 +186,127 @@ public function deletePersistentSubscriptionAsync(
     ): Promise;
 
     /**
+     * @param string $stream
+     * @param bool $resolveLinkTos
+     * @param Closure(EventStoreSubscription, ResolvedEvent): Promise $eventAppeared
+     * @param ?Closure(EventStoreSubscription, SubscriptionDropReason, ?Throwable): void $subscriptionDropped
+     * @param ?UserCredentials $userCredentials
      * @return Promise<EventStoreSubscription>
      */
     public function subscribeToStreamAsync(
         string $stream,
         bool $resolveLinkTos,
-        EventAppearedOnSubscription $eventAppeared,
-        ?SubscriptionDropped $subscriptionDropped = null,
+        Closure $eventAppeared,
+        ?Closure $subscriptionDropped = null,
         ?UserCredentials $userCredentials = null
     ): Promise;
 
     /**
+     * @param string $stream
+     * @param ?int $lastCheckpoint
+     * @param ?CatchUpSubscriptionSettings $settings
+     * @param Closure(EventStoreCatchUpSubscription, ResolvedEvent): Promise $eventAppeared
+     * @param ?Closure(EventStoreCatchUpSubscription): void $liveProcessingStarted
+     * @param ?Closure(EventStoreCatchUpSubscription, SubscriptionDropReason, ?Throwable): void $subscriptionDropped
+     * @param ?UserCredentials $userCredentials
```

Comment:
Do PhpStorm, PHPStan and Psalm all understand `?` in annotations instead of `|null`?

Reply:
By the way I heavily dislike duplicate annotations that do not add anything that's not already known from the typehints. So here I'd drop all the param annotations except the closures.

Reply:
> Do PhpStorm, PHPStan and Psalm all understand `?` in annotations instead of `|null`?

phpstorm - no (maybe revert?)
psalm - yes
phpstan - i don't know

Reply:
> By the way I heavily dislike duplicate annotations that do not add anything that's not already known from the typehints. So here I'd drop all the param annotations except the closures.

okay.